pulumi/infra/lambda_.py

```diff
@@ -173,10 +173,10 @@ def __init__(
             opts=pulumi.ResourceOptions(parent=self),
         )
 
-        lambda_name = f"{DEPLOYMENT_NAME}-{name}"
+        lambda_name = override_name or f"{DEPLOYMENT_NAME}-{name}"
         self.function = aws.lambda_.Function(
             f"{name}-lambda",
-            name=override_name or name,
+            name=lambda_name,
```

Comment:
This is still explicitly setting a name for everything... it would need to be `name = override_name or None`